tensorflow/compiler/mlir/hlo/runtime/ral/ral_logging_test.cc

```diff
@@ -20,15 +20,22 @@ limitations under the License.
 
 namespace testing {
 
+using ::mlir::disc_ral::ERROR;
+using ::mlir::disc_ral::FATAL;
+using ::mlir::disc_ral::INFO;
+using ::mlir::disc_ral::WARNING;
+using ::mlir::disc_ral::internal::LogMessage;
+
 TEST(Logging, Log) {
-  DISC_LOG(INFO) << "Hello";
-  DISC_LOG(INFO) << "Another log message";
-  DISC_LOG(ERROR) << "Error message";
-  DISC_VLOG(1) << "A VLOG message";
-  DISC_VLOG(2) << "A higher VLOG message";
-  if (DISC_VLOG_IS_ON(1)) {
-    DISC_VLOG(1) << "A DISC_VLOG_IS_ON message";
-  }
+  EXPECT_TRUE(DISC_VLOG_IS_ON(0));
+  EXPECT_FALSE(DISC_VLOG_IS_ON(1));
+
+  LogMessage logger(__FILE__, __LINE__, ERROR);
+  logger << "Hello";
+  EXPECT_TRUE(logger.GetFilterStringForTesting(INFO) == "Hello");
+  EXPECT_TRUE(logger.GetFilterStringForTesting(WARNING) == "Hello");
+  EXPECT_TRUE(logger.GetFilterStringForTesting(ERROR) == "Hello");
```

Comment:
Looks like errors are swallowed by default?

Reply:
This is because the severity setting of the logger (Line 33). `FATAL` has the highest severity and `INFO` has the lowest severity. The message is print only when the severity of the logger is larger than `min_log_level` (specified by the first argument of function `GetFilterStringForTesting `).  The last one (Line 38) is not print because `min_log_level (FATAL)` is larger than the severity of the logger (`ERROR`).

Reply:
I understand this: I'm questioning whether it is good that by default `LOG(ERROR)` would be silent?

Reply:
The default value of `min_log_level ` is `0`(Line 44, file `ral_logging.cc`). Thus, logger with severity large than `0` (INFO) will be print by default (Line 140, file file `ral_logging.cc`).

Reply:
`VLOG` is different from `LOG` and is only used for debug. It does not respect usually severity as `LOG`. The larger the number of `VLOG` is, the less important the message is.